src/client/Microsoft.Identity.Client/Internal/Broker/BrokerInteractiveRequestComponent.cs

```diff
@@ -108,5 +117,40 @@ internal void ValidateResponseFromBroker(MsalTokenResponse msalTokenResponse)
                 throw new MsalServiceException(MsalError.BrokerResponseReturnedError, MsalErrorMessage.BrokerResponseReturnedError, null);
             }
         }
+
+        // Example auth code that shows that broker is required:
+        // msauth://wpj?username=joe@contoso.onmicrosoft.com&app_link=itms%3a%2f%2fitunes.apple.com%2fapp%2fazure-authenticator%2fid983156458%3fmt%3d8
+        public static bool IsBrokerRequiredAuthCode(string authCode, out string installationUri)
+        {
+            if (authCode.StartsWith(BrokerParameter.AuthCodePrefixForEmbeddedWebviewBrokerInstallRequired, StringComparison.OrdinalIgnoreCase))
+            //|| authCode.StartsWith(_serviceBundle.Config.RedirectUri, StringComparison.OrdinalIgnoreCase) // TODO: what is this?!
+            {
+                installationUri = ExtractAppLink(authCode);
+                return (installationUri != null);
+            }
+
+            installationUri = null;
+            return false;
+        }
+
+        private static string ExtractAppLink(string authCode)
+        {
+            Uri authCodeUri = new Uri(authCode);
+            string query = authCodeUri.Query;
+
+            if (query.StartsWith("?", StringComparison.OrdinalIgnoreCase))
+            {
+                query = query.Substring(1);
+            }
+
+            Dictionary<string, string> queryDict = CoreHelpers.ParseKeyValueList(query, '&', true, true, null);
+
+            if (!queryDict.ContainsKey(BrokerParameter.AppLink))
+            {
+                return null;
```

Comment:
@jennyf19 - is it possible for auth code to start with msauth// but not contain app link? Should we throw in this case instead?